Sources/OpenTelemetry/Tracing/Processing/Batch/OTelBatchSpanProcessorConfiguration.swift

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2023 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// The configuration options for an ``OTelBatchSpanProcessor``.
+public struct OTelBatchSpanProcessorConfiguration: Sendable {
+    /// The maximum queue size.
+    ///
+    /// - Warning: After this size is reached spans will be dropped.
+    public var maximumQueueSize: UInt
+
+    /// The maximum delay between two consecutive span exports.
+    public var scheduleDelay: Duration
+
+    /// The maximum batch size of each export.
+    ///
+    /// - Note: If the queue reaches this size, a batch will be exported even if ``scheduleDelay`` has not elapsed.
+    public var maximumExportBatchSize: UInt
+
+    /// The duration a single export can run until it is cancelled.
+    public var exportTimeout: Duration
+
+    /// Create a batch span processor configuration.
+    ///
+    /// - Parameters:
+    ///   - environment: The environment variables.
+    ///   - maximumQueueSize: A maximum queue size used even if `OTEL_BSP_MAX_QUEUE_SIZE` is set. Defaults to `2048` if both are `nil`.
+    ///   - scheduleDelay: A schedule delay used even if `OTEL_BSP_SCHEDULE_DELAY` is set. Defaults to `5` seconds if both are `nil`.
+    ///   - maximumExportBatchSize: A maximum export batch size used even if `OTEL_BSP_MAX_EXPORT_BATCH_SIZE` is set. Defaults to `512` if both are `nil`.
+    ///   - exportTimeout: An export timeout used even if `OTEL_BSP_EXPORT_TIMEOUT` is set. Defaults to `30` seconds if both are `nil`.
+    public init(
+        environment: OTelEnvironment,
+        maximumQueueSize: UInt? = nil,
+        scheduleDelay: Duration? = nil,
+        maximumExportBatchSize: UInt? = nil,
+        exportTimeout: Duration? = nil
+    ) {
+        self.maximumQueueSize = environment.requiredValue(
+            programmaticOverride: maximumQueueSize,
+            key: "OTEL_BSP_MAX_QUEUE_SIZE",
```

Comment:
Can we store those env keys in an internal struct somewhere?

Reply:
Since the environment variables are used in different targets I'd not group them in a single struct, and I think having smaller structs within the individual configuration types to hold their environment variables is overkill since they're not re-used elsewhere.